src/rpc_client/mod.rs

```diff
@@ -306,12 +320,22 @@ impl fmt::Display for Url {
     }
 }
 
+// The communication protocol
+enum ComProtocol {
+    Http,
+    Ws,
+}
+
 /// Parses a multi-address into a URL
-fn multiaddress_to_url(multiaddr: &Multiaddr, endpoint: &str) -> Url {
+fn multiaddress_to_url(multiaddr: &Multiaddr, endpoint: &str, com_protocol: ComProtocol) -> Url {
     // Fold Multiaddress into a Url struct
     let addr = multiaddr.iter().fold(
         Url {
-            protocol: DEFAULT_PROTOCOL.to_owned(),
+            protocol: match com_protocol {
+                ComProtocol::Http => HTTP_PROTOCOL,
```

Comment:
Do we or are we going to support `https` and `wss`?

Reply:
We don't support `wss` right now.
And I don't think we are supporting https, either.

We might support them in the future though.